src/scripts/main.js

```diff
@@ -1,3 +1,60 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const buttons = document.querySelectorAll('button');
+
+buttons.forEach((button) => {
+  button.addEventListener('click', (e) => {
+    const tr = document.querySelectorAll('tr');
+
+    const tdCount = tr[0] ? tr[0].querySelectorAll('td').length : 0;
+
+    if (tr.length <= 2) {
+      document.querySelector('.remove-row.button').disabled = true;
+    } else {
+      document.querySelector('.remove-row.button').disabled = false;
+    }
+
+    if (tdCount <= 2) {
+      document.querySelector('.remove-column.button').disabled = true;
+    } else {
+      document.querySelector('.remove-column.button').disabled = false;
+    }
+
+    if (tr.length >= 10) {
+      document.querySelector('.append-row.button').disabled = true;
+    } else {
+      document.querySelector('.append-row.button').disabled = false;
+    }
+
+    if (tdCount >= 10) {
+      document.querySelector('.append-column.button').disabled = true;
+    } else {
+      document.querySelector('.append-column.button').disabled = false;
+    }
+
+    if (e.target.className === 'append-row button') {
+      const firstTr = tbody.querySelector('tr');
+
+      firstTr.before(firstTr.cloneNode(true));
```

Comment:
Cloning the first row and inserting it before itself might not be ideal if the first row contains headers or specific data. Consider appending a new row at the end or using a template row for cloning.